packages/vm/lib/index.ts

```diff
@@ -209,14 +210,14 @@ export default class VM extends AsyncEventEmitter {
     const { opts } = this
 
     if (opts.activatePrecompiles && !opts.stateManager) {
-      this.stateManager.checkpoint()
+      await this.stateManager.checkpoint()
```

Comment:
Whew, we did forget so many await (the other one e.g. what @cgewecke mentioned)? 😕

This could lead to serious unpredictable and buggy behavior, right? Is anything so apparently leading to potential malfunction here that it should be mentioned in the release notes?

Reply:
The other one seems a bit more alarming, but they could definitely cause some race conditions, which might be worth mentioning, i.e. fix some possible race conditions.